swig/include/python/ogr_python.i

```diff
@@ -482,6 +482,214 @@ def ReleaseResultSet(self, sql_lyr):
 
   %}
 
+  %feature("shadow") SetAttributeFilter %{
+  def SetAttributeFilter(self, filter_string: "Optional[str]"):
+      """SetAttributeFilter(self, filter_string: Optional[str])
+
+      Set the attribute query string to be used when fetching features.
+
+      See :cpp:func:`OGRLayer::SetAttributeFilter` for more information on
```

Comment:
Seems a little weird to put RST stuff in the docstrings, but I'm not sure of another way to make this link.